dali/kernels/erase/erase_gpu_test.cu

```diff
@@ -0,0 +1,390 @@
+
+    // cudaEventRecord(end);
+    CUDA_CALL(cudaDeviceSynchronize());
+    CUDA_CALL(cudaGetLastError());
+
+    // float t = 0;
+    // cudaEventElapsedTime(&t, start, end);
+    // int64_t read =
+    //     in_view.num_elements() * sizeof(T) + regions_gpu.num_elements() * sizeof(ibox<ndim>);
+    // int64_t written = out_view.num_elements() * sizeof(T);
+    // int64_t total_transferred = read + written;
+
+    // std::cerr << "Transferred " << total_transferred << " in " << t << "[ms] with "
+    //           << total_transferred / t * 1e-6 << " GB/s" << endl;
+
+    RepackAndCalcCpu();
+    Verify();
+  }
+
+  void Verify() {
+    auto cpu_out_view = output.cpu();
+    auto cpu_baseline_view = baseline.cpu();
+    Check(cpu_out_view, cpu_baseline_view);
+  }
+
+  void RepackAndCalcCpu() {
+    auto input_tlv = input.cpu();
+    auto baseline_tlv = baseline.cpu();
+    auto regions_tlv = regions.cpu();
+    for (int i = 0; i < batch_size; i++) {
+      auto baseline_tv = baseline_tlv[i];
+      auto input_tv = input_tlv[i];
+
+      EraseArgs<float, ndim> args;
+      args.rois.resize(num_erase_regions_);
+      for (int j = 0; j < num_erase_regions_; j++) {
+        for (int d = 0; d < ndim; d++) {
+          args.rois[j].anchor[d] = regions_tlv[i](j)->lo[d];
+          args.rois[j].shape[d] = regions_tlv[i](j)->hi[d] - regions_tlv[i](j)->lo[d];
+          args.rois[j].fill_values = fill_values;
+          args.rois[j].channels_dim = channel_dim;
+        }
+      }
+
+      EraseCpu<T, ndim> cpu_kernel;
+      KernelContext ctx;
+      cpu_kernel.Run(ctx, baseline_tv, input_tv, args);
+    }
+  }
+
+  void CreateRegions() {
+    if (region_generation_ == 0) {
+      num_erase_regions_ = 0;
+      // no cover
+    } else if (region_generation_ == 1) {
+      // full cover
+      num_erase_regions_ = 1;
+    }
+    TensorListShape<1> regions_shape = uniform_list_shape<1>(batch_size, {num_erase_regions_});
+    regions.reshape(regions_shape);
+    auto regions_cpu = regions.cpu();
+    if (region_generation_ == 1) {
+      // full cover
+      for (int i = 0; i < batch_size; i++) {
+        auto regions_tv = regions_cpu[i];
+        *regions_tv(0) = ibox<ndim>({0}, to_ivec(shape_));
+      }
+    } else if (region_generation_ == 2) {
+      std::mt19937 gen(0);
+      for (int i = 0; i < batch_size; i++) {
+        auto regions_tv = regions_cpu[i];
+        for (int j = 0; j < num_erase_regions_; j ++) {
+          ibox<ndim> region_box;
+          for (int d = 0; d < ndim; d++) {
+            std::uniform_int_distribution<>  start_dim(0, shape_[d] - 1);
+            region_box.lo[d] = start_dim(gen);
+            std::uniform_int_distribution<>  end_dim(region_box.lo[d] + 1, shape_[d]);
+            region_box.hi[d] = end_dim(gen);
+          }
+          *regions_tv(j) = region_box;
+        }
+      }
+    }
+  }
+
+  int num_erase_regions_;
+  int region_generation_;
+  bool full_fill_;
+  std::vector<T> fill_values;
```

Comment:
```suggestion
  std::vector<T> fill_values_;
```

Reply:
You should do this for `test_shape`, `batch_size` , input`, `output`, `baseline` and `regions` as well.

Reply:
Done